src/NuGet.Core/NuGet.ProjectModel/JsonUtility.cs

```diff
@@ -6,13 +6,19 @@
 using System.IO;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
+using NuGet.Common;
 using NuGet.Packaging.Core;
 using NuGet.Versioning;
 
 namespace NuGet.ProjectModel
 {
     internal static class JsonUtility
     {
+        private static readonly Dictionary<string, NuGetVersion> NuGetVersionCache = new();
+        private static readonly Dictionary<string, VersionRange> VersionRangeCache = new();
```

Comment:
This will leak memory. Every version and version range ever parsed during the entire lifetime of the process will be retained.

Reply:
Yea I'm not super stocked on this solution. I asked in [this comment](https://github.com/NuGet/NuGet.Client/pull/5558#discussion_r1448182312) if we should consider a specific lifetime for these dictionaries.

Reply:
fwiw, my recommendation in that thread was to make the lifetime an assets file read, not a static inside the class.

Reply:
yea I think that scope makes the most sense.

Reply:
Undid this change per Andy's suggestion to reduce scope creep.